crates/ra_hir/src/diagnostics.rs

```diff
@@ -113,3 +113,31 @@ impl Diagnostic for UnresolvedModule {
         self
     }
 }
+
+#[derive(Debug)]
+pub struct FillStructFields {
+    pub file: HirFileId,
+    pub decl: AstPtr<ast::NamedFieldList>,
```

Comment:
`Declaration` is for things that define an item (struct decl, fn decl, etc). I'd just use `field_list` for this